pkg/feeds/feed.go

```diff
@@ -0,0 +1,73 @@
+package feeds
+
+import (
+	"encoding"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethersphere/bee/pkg/crypto"
+	"github.com/ethersphere/bee/pkg/soc"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+type id struct {
+	topic []byte
+	index []byte
+}
+
+func (i *id) MarshalBinary() ([]byte, error) {
+	return crypto.LegacyKeccak256(append(i.topic, i.index...))
+}
+
+// Feed is representing an epoch based feed
+type Feed struct {
+	Topic []byte
+	Owner common.Address
+}
+
+// New constructs an epoch based feed from a human readable topic and an ether address
+func New(topic string, owner common.Address) (*Feed, error) {
+	th, err := crypto.LegacyKeccak256([]byte(topic))
+	if err != nil {
+		return nil, err
+	}
+	return &Feed{th, owner}, nil
+}
+
+type Index interface {
```

Comment:
not entirely sure why this interface is needed

Reply:
because, feeds generalise over indexing schemes, both epochs.epoch and sequence.index implement this interface